### ISSUE TITLE
Loss Function and learning rate

### ISSUE THREAD

**Issue by htjb** [htjb/margarine#14]
It would be useful for users to be able to set the learning rate to a learning rate schedule as we find that the loss is not always smooth.

For example, when learning the following gaussian distribution in blue

<img src="https://user-images.githubusercontent.com/40355093/222961723-d32b1297-ac3e-4ad1-97d9-915e33bc474d.png" height=300>

for 500 epochs and sampling the autoregressive flow, shown in orange, the loss function is not smooth

<img src="https://user-images.githubusercontent.com/40355093/222961733-a4e20fc0-0f94-4185-b758-fa4c5fb3dbca.png" height=300>

We find that using a learning rate scheduler can help alleviate this. Since the particular choice of learning rate or learning rate schedule is problem specific, I would suggest changing the following requirement on the `learning_rate` kwarg in `maf.py` from

```python
if type(self.learning_rate) not in [int, float]:
    raise TypeError("'learning_rate', must be a float.")
```

to 

```python
if not isinstance(lr_schedule, 
                  (int, float, 
                   keras.optimizers.schedules.LearningRateSchedule)):
    raise TypeError("'learning_rate', " + 
                    "must be an integer, float or keras scheduler.")
```

Training on the same distribution with the following learning rate scheduler

```python
lr_schedule = keras.optimizers.schedules.ExponentialDecay(
    initial_learning_rate=1e-3,
    decay_steps=25,
    decay_rate=0.9)
```

gives the following results

<img src="https://user-images.githubusercontent.com/40355093/222961931-258b85fe-9b03-4617-a85e-f7b0ab74f53a.png" height=300>

with a smoother loss history

<img src="https://user-images.githubusercontent.com/40355093/222961820-243115b4-be38-4e69-b09e-25668ff5b279.png" height=300>

**Comment by htjb** [comment]
Closed by #15
